lua/refactoring/refactor/106.lua

```diff
@@ -64,6 +64,18 @@ local function get_extract_setup_pipeline(bufnr)
         :add_task(get_selected_local_defs)
 end
 
+local function get_start_col(str)
```

Comment:
Perhaps all of this white space logic could be moved into a class that is a bit more fitting.

Meaning you give it a piece of code, it could even find the scope and try to determine some things about it.  But for now it could encapsulate this logic.

Reply:
I like the idea of moving this to a class. Ideally I wanted to pass in the node for this block, then get the indentation of the node, but I couldn't find the right function in the treesitter docs.

Reply:
Moved code to it's own class, let me know if there's anything else I should add.

Reply:
there is also `'indentexpr'` which evaluates the line for the "proper" indent.  We could start using this. 

I know @tjdevries talked about it last night.

Reply:
So got something working using `indentexpr` on [this branch](https://github.com/teddylear/refactoring.nvim/compare/feature/python-indenting...teddylear:feature/python-indenting-indentexpr?expand=1), but the issue is that the default indentexpr in neovim `GetPythonIndent(v:lnum)` does not indent the lines correct. To test this run `:echo GetPythonIndent(4)` on the start.py file for 106 (it will return 0). I did have this working locally with [this plugin](https://github.com/Vimjas/vim-python-pep8-indent/blob/master/indent/python.vim), but this would have to be added as a plugin dependency. Testing this with `:echo GetPythonPEPIndent(4)` which will result in 4. Let me know if you want me to continue down the `indentexpr` avenue or where you think I should go from here.